src/main/io/rcdevice.c

```diff
@@ -364,41 +359,52 @@ static rcdeviceResponseParseContext_t* getWaitingResponse(timeMs_t currentTimeMs
+    while ((respCtx = getWaitingResponse(millis())) != NULL) {
+        if (!serialRxBytesWaiting(respCtx->device->serialPort)) {
+            break;
+        }
 
+        const uint8_t c = serialRead(respCtx->device->serialPort);
         if (respCtx->recvRespLen == 0) {
             // Only start receiving packet when we found a header
-            if ((respCtx->protocolVer == RCDEVICE_PROTOCOL_VERSION_1_0 && c != RCDEVICE_PROTOCOL_HEADER) || (respCtx->protocolVer == RCDEVICE_PROTOCOL_RCSPLIT_VERSION && c != RCSPLIT_PACKET_HEADER)) {
+            if ((respCtx->protocolVersion == RCDEVICE_PROTOCOL_VERSION_1_0 && c != RCDEVICE_PROTOCOL_HEADER) || (respCtx->protocolVersion == RCDEVICE_PROTOCOL_RCSPLIT_VERSION && c != RCSPLIT_PACKET_HEADER)) {
                 continue;
             }
         }
-        
+
         respCtx->recvBuf[respCtx->recvRespLen] = c;
         respCtx->recvRespLen += 1;
-        
+
         // if data received done, trigger callback to parse response data, and update rcdevice state
         if (respCtx->recvRespLen == respCtx->expectedRespLen) {
-            // verify the crc value
-            if (respCtx->protocolVer == RCDEVICE_PROTOCOL_VERSION_1_0) {
+            if (respCtx->protocolVersion == RCDEVICE_PROTOCOL_VERSION_1_0) {
                 uint8_t crc = 0;
                 for (int i = 0; i < respCtx->recvRespLen; i++) {
                     crc = crc8_dvb_s2(crc, respCtx->recvBuf[i]);
                 }
-                
+
                 respCtx->result = (crc == 0) ? RCDEVICE_RESP_SUCCESS : RCDEVICE_RESP_INCORRECT_CRC;
-            } else if (respCtx->protocolVer == RCDEVICE_PROTOCOL_RCSPLIT_VERSION) {
-                // do nothing, just call parserFunc
-                respCtx->result = RCDEVICE_RESP_SUCCESS;
+            } else if (respCtx->protocolVersion == RCDEVICE_PROTOCOL_RCSPLIT_VERSION) {
+                if (respCtx->recvBuf[0] == RCSPLIT_PACKET_HEADER && respCtx->recvBuf[1] == RCSPLIT_PACKET_CMD_CTRL && respCtx->recvBuf[2] == 0xFF && respCtx->recvBuf[4] == RCSPLIT_PACKET_TAIL) {
+                    uint8_t crcFromPacket = respCtx->recvBuf[3];
+                    respCtx->recvBuf[3] = respCtx->recvBuf[4]; // move packet tail field to crc field, and calc crc with first 4 bytes
+                    uint8_t crc = crc8HighFirst(respCtx->recvBuf, 4);
+                    
+                    respCtx->result = crc == crcFromPacket ? RCDEVICE_RESP_SUCCESS : RCDEVICE_RESP_INCORRECT_CRC;
+                } else {
+                    respCtx->result = RCDEVICE_RESP_INCORRECT_CRC;
+                }
             }
 
             if (respCtx->parserFunc != NULL) {
                 respCtx->parserFunc(respCtx);
             }
 
-            // dequeue current response context
-            rcdeviceRespCtxQueueShift(&watingResponseQueue);
+            if (respCtx->result == RCDEVICE_RESP_SUCCESS) {
+                rcdeviceRespCtxQueueShift(&watingResponseQueue);
```

Comment:
You could fix a spelling error here as well ("waiting").

Reply:
👍